### PR TITLE
[API][SIMS-#1248] API Client Naming

### PR DESCRIPTION
**Pull request by henryoforeh-aot**
- Changed API sub resource(s) from plural to singular.
- Updated affected files due to updated subresources in vue web application.
- Updated wiki documentation.

**Wiki Update**
https://github.com/bcgov/SIMS/wiki/Development-Standards-and-Decisions#controller-api-naming-convention
image

### CODE REVIEW

**Review by ann-aot — APPROVED** [review]
👍

File: sources/packages/web/src/services/http/StudentApi.ts [hunk 1]
```diff
@@ -24,7 +24,7 @@ export class StudentApi extends HttpBaseClient {
    * @param studentProfile information needed to create the user.
    */
   async createStudent(studentProfile: CreateStudentAPIInDTO): Promise<void> {
-    await this.postCall(this.addClientRoot("students"), studentProfile);
+    await this.postCall(this.addClientRoot("student"), studentProfile);
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/services/http/StudentApi.ts:27]: Just to every body be on the same page, this was what @dheepak-aot mention about having the const.
We would be creating a const like below and reuing among all app calls. Dheepak, correct me if I am wrong.
```ts
const API_CONTROLLER =  "student";
```

```ts
await this.postCall(this.addClientRoot(API_CONTROLLER), studentProfile);
return this.postCall(this.addClientRoot(`${API_CONTROLLER}/search`), payload);
````

**Review by andrewsignori-aot — COMMENTED** [review]
The changes look good 👍 
I believe that we are missing also the `aest/reports`, right?

File: sources/packages/web/src/services/http/StudentApi.ts (hunk 1 above)
Comment (henryoforeh-aot) [reply sources/packages/web/src/services/http/StudentApi.ts:27]: maybe I can just make the change in this PR so that we don't forget about it.

**Review by guru-aot — APPROVED** [review]
LGTM

**Review by andrewsignori-aot — APPROVED** [review]
Looks good, thanks for doing this changes also 👍

**Pull request merged**